src/components/Popups/PopupItem.tsx

```diff
@@ -45,32 +47,32 @@ export default function PopupItem({
   popKey: string
 }) {
   const removePopup = useRemovePopup()
-  const removeThisPopup = useCallback(() => removePopup(popKey), [popKey, removePopup])
+  const theme = useTheme()
+
   useEffect(() => {
     if (removeAfterMs === null) return undefined
 
     const timeout = setTimeout(() => {
-      removeThisPopup()
+      removePopup(popKey)
     }, removeAfterMs)
 
     return () => {
       clearTimeout(timeout)
     }
-  }, [removeAfterMs, removeThisPopup])
-
-  const theme = useTheme()
+  }, [popKey, removeAfterMs, removePopup])
+  const isTxnPopup = 'txn' in content
 
   let popupContent
-  if ('txn' in content) {
+  if (isTxnPopup) {
     popupContent = <TransactionPopup hash={content.txn.hash} />
   } else if ('failedSwitchNetwork' in content) {
     popupContent = <FailedNetworkSwitchPopup chainId={content.failedSwitchNetwork} />
   }
 
-  return popupContent ? (
-    <Popup>
-      <StyledClose color={theme.textSecondary} onClick={removeThisPopup} />
+  return (
+    <Popup isTransactionPopup={isTxnPopup} show={!!popupContent}>
```

Comment:
same with `StyledClose` - it seems like the only shared property there is the hover state?

Reply:
actually i'd probably just refactor this whole component to just return early on content type instead of shoving the type into a variable and toggling each of the react tree items depending on the variable

if ('txn' in content) {
   return <TransactionPopupStuff />
}

else return <RegularPopupStuff />

Reply:
great suggestion thanks, refactor into separate styled components re-using the same shared CSS stuff. I think it looks much cleaner now.